### DISCUSSION TITLE
How to configure retention policy for block storage?

### DISCUSSION THREAD

**Discussion by zarom** [cortexproject/cortex#3631] [Q&A] [answered]
Is it possible to configure retention policy for block storage? For example, if I want to store metrics for 6 months only.

**Comment by pracucci** [comment, marked as answer]
Currently, the retention policy for Cortex running with the blocks storage can be enforced at the object store (typically with a bucket policy, eg. when running in the cloud).

_We also plan to add per-tenant custom retention support in the compactor, but the work hasn't started yet._

**Comment by lzh-lab** [comment]
I think it can be divided into two steps, first implement a global retention strategy, this is simple and rewarding.

After that, add the per-tenant custom retention, the retention of per-tenant must be less than the global retention.

**Reply by pracucci** [reply]
The global retention can already be implemented with a bucket policy. I think the design doc we'll work on should focus on per-tenant custom retention. Shouldn't be much more difficult (actually, I believe it will be same complexity).
